### PR TITLE
Parameters Support for List And Map

### PR DESCRIPTION
**Pull request by anandsubbu007**
Now we can able to pass List of String, double, int, DateTime, model (that generated), bool as parameter in endpoint

Map<String, dynamic> parameter can also be passed as an argument.

with return of list

### CODE REVIEW

**Comment by anandsubbu007** [comment]
Client Side
```
      final strTst = await client.userFunc.testListOfString(['Hi', 'There']);
      print(strTst);
      final bolTst = await client.userFunc.testListOfbool([true, false]);
      print(bolTst);
      final dobTst = await client.userFunc.testListOfdouble([1, 2, 3.3]);
      print(dobTst);
      final dynamTst =
          await client.userFunc.testListOfdynamic(['String txt', 123]);
      print(dynamTst);
      final mapTst = await client.userFunc
          .testListOfmapDynamic({'Name': 'Anand', "Age": 24});
      print(mapTst);
      final modelTst = await client.userFunc.testListOfSimpleData([Example(name:'Anand',data: 007 )]);
      print(modelTst);
      final dtTst = await client.userFunc
          .testListOfDateTime([DateTime(2020), DateTime(2020)]);
      print(dtTst);
```

Server Side Code

```
  Future<List<String>> testListOfString(
      Session session, List<String> value) async {
    print('Received: $value');
    return value;
  }

  Future<List> testListOfdynamic(Session session, List value) async {
    print('Received: $value');
    return value;
  }

  Future<List<double>> testListOfdouble(
      Session session, List<double> value) async {
    print('Received: $value');
    return value;
  }

  Future<List<DateTime>> testListOfDateTime(
      Session session, List<DateTime> value) async {
    print('Received: $value');
    return value;
  }

  Future<List<bool>> testListOfbool(Session session, List<bool> value) async {
    print('Received: $value');
    return value;
  }

  Future<Map<String, dynamic>> testListOfmapDynamic(
      Session session, Map<String, dynamic> value) async {
    print('Received: $value');
    return value;
  }

  Future<List<BannersClass>> testListOfSimpleData(
      Session session, List<Example> object) async {
    int i = 0;
    for (var e in object) {
      i++;
      e.data = i.toString();
    }
    return object;
  }

```

File: modules/serverpod_chat/serverpod_chat_flutter/pubspec.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: serverpod_chat_flutter
 description: A new Flutter package project.
 version: 0.0.1
-homepage:
```
Comment (vlidholt) [inline modules/serverpod_chat/serverpod_chat_flutter/pubspec.yaml:4]: Make sure to not include changes that aren't relevant to the feature

**Comment by vlidholt** [comment]
Before I review this pull request I just wanted to check some general things.

Does it support Maps with any of the Serverpod primitive types (int, bool, double, DateTime, ByteData) both as keys and values? Does it support Serializable as key and value? Are there tests for all of the types?

Currently the tests are failing for the PR. It's possible to run the tests locally before doing a PR by doing `util/run_tests` (unfortunately may only work on Mac yet).

**Review by ordru — COMMENTED** [review]
- If I am not mistaken, the destination branch should be `dev` instead of `main`?

**Comment by anandsubbu007** [comment]
> util/run_tests

It Supports
List of String, int, DateTime, bool, double, SerializableEntity & dynamic(if within this mentioned types)
For Map
Keys -> String 
Values -> String, double, int, bool, DateTime, SerializableEntity (with null support)


I didn't test it with byteData & other key types value in Map

**Comment by anandsubbu007** [comment]
> util/run_tests

Can't able to run Test in my mac
```
The terminal process "/bin/zsh '--login'" terminated with exit code: 1.
```

**Comment by anandsubbu007** [comment]
All test passed in my device. 
with dart 17 sdk.
Let me know if i done something wrong

**Comment by vlidholt** [comment]
I'm closing this as it doesn't support e.g. Map<DateTime, String?> or Map<int, dynamic>. I'm planning a bit of another approach to support these and also any of the supported types as parameters.

In future, make sure to also send requests to the dev branch.

**Pull request closed without merging**